### PR TITLE
Hera V4

### PR DESCRIPTION
**Pull request by flaviuvadan**
## TODO

- [x] docstrings
- [x] examples
- [x] unit tests
- [x] functional tests
- [x] dependency check can lead to false positive results, `TODO:task.py:260`
- [x] ~`when` (decided to move ahead with examples, study the ergonomics there, and then revisit, deprioritized)~

Adds:
- [x] #88 - workflow of workflows and patching support
- [x] #177 - fan out over multiple task implementations through `InputFrom`
- [x] #218 - pod patch spec via parameterization

Breaking changes:
- [x] #201 - expose namespace configuration on workflow services only rather than workflows
- [x] #202 215 - support for arbitrary script submissions via `source` rather than `func`
- [x] #205 - make `Resource` more aligned with the usage and specifications in Kubernetes
- [x] #220 - dynamic allocation and exposure of task and workflow output fields

Discussion topic:
- [x] #204 - should we rename some objects so that they are more in line with Argo specifications and allow control over dynamic field reference via `{{...}}` syntax? CC: @trollgeir

### CODE REVIEW

File: src/hera/resources.py [hunk 1]
```diff
@@ -70,15 +70,20 @@ def __post_init__(self):
...
+        if self.memory_request is None and self.memory_limit is not None:
+            self.memory_request = self.memory_limit
+
     def build(self) -> ResourceRequirements:
         """Builds the resource requirements of the pod"""
         resources = dict()
 
         if self.cpu_limit is not None:
-            resources = merge_dicts(resources, dict(limit=dict(cpu=self.cpu_limit)))
+            resources = merge_dicts(resources, dict(limit=dict(cpus=str(self.cpu_limit))))
```
Comment (JacobHayes) [inline src/hera/resources.py:83]: Should this one be `cpus` plural (but singular below for request)?

**Comment by JacobHayes** [comment]
🎉

File: src/hera/resources.py (hunk 1 above)
Comment (flaviuvadan) [reply src/hera/resources.py:83]: Independent of anything else yes, I agree with you, but recall that `ResourceRequirements` is mapped to Kubernetes resource requirements, and those use `cpu` only, inside the `requests` and `limits` fields! See [docs here](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-units-in-kubernetes)!

Comment (JacobHayes) [reply src/hera/resources.py:83]: In that case, should this be:
```suggestion
            resources = merge_dicts(resources, dict(limit=dict(cpu=str(self.cpu_limit))))
```
?

Comment (flaviuvadan) [reply src/hera/resources.py:83]: Oh... yes 😬

**Pull request merged**